src/AccountIcon.jsx

```diff
@@ -1,45 +1,50 @@
 const blockies = require('blockies');
 const React = require('react');
-const {Image} = require('semantic-ui-react');
-const {ReactiveComponent} = require('oo7-react');
+const PropTypes = require('prop-types');
+
+const { Image } = require('semantic-ui-react');
+const { ReactiveComponent } = require('oo7-react');
 
 function createIdentityImage (address, scale = 8) {
 	return blockies({
-    	seed: (address || '').toLowerCase(),
-    	size: 8,
-    	scale
-    }).toDataURL();
+		seed: (address || '').toLowerCase(),
+		size: 8,
+		scale
+	}).toDataURL();
 }
 
 class AccountIcon extends ReactiveComponent {
-	constructor() { super(['address', 'className', 'style']); }
+	constructor () { super(['address', 'className', 'style']); }
 
-	render() {
-		if (typeof(this.state.address) == "string") {
-			return (<Image
-				inline
-				avatar={this.props.avatar}
-                src={createIdentityImage(this.state.address)}
-				style={Object.assign({borderRadius: '50%'}, this.state.style)}
-                className={typeof(this.state.className) === 'string' ? this.state.className : ''}
-                id={this.props.id}
-                data-address-img
-            />);
+	render () {
+		if (typeof (this.state.address) === 'string') {
```

Comment:
You're using a lot of typeof in your code. I think it should really be used for variables that can have different meaningful types (like string & number). Correct me if I'm wrong, but here address can only be string (or undefined), right?

If this case this whole block would look better with `return this.state.address ? (...) : (...)`

My advice would be to use typeof as little as possible. Look everywhere is your code, and only use typeof if it's really necessary.

Reply:
I believe its being used here to validate that the input(again, didnt make this one ;) ), since this library is intended to be used to make who knows what, if a developer makes an error and a wrong type comes in you might want some kind of fallback no?